internal/replacements/replacements.go

```diff
@@ -94,9 +70,43 @@ func ReplaceMisconfiguredProcessGroups(ctx context.Context, podManager podmanage
 	return hasReplacements, nil
 }
 
+// ProcessGroupNeedsRemoval checks if a process group needs to be removed.
+func ProcessGroupNeedsRemoval(ctx context.Context, podManager podmanager.PodLifecycleManager, client client.Client, log logr.Logger, cluster *fdbv1beta2.FoundationDBCluster, processGroup *fdbv1beta2.ProcessGroupStatus, pvcMap map[fdbv1beta2.ProcessGroupID]corev1.PersistentVolumeClaim) (bool, error) {
+	// TODO(johscheuer): Fix how we fetch the pvc to make better use of the controller runtime cache.
+	pvc, hasPVC := pvcMap[processGroup.ProcessGroupID]
+	pod, podErr := podManager.GetPod(ctx, client, cluster, processGroup.GetPodName(cluster))
+	if hasPVC {
+		needsPVCRemoval, err := processGroupNeedsRemovalForPVC(cluster, pvc, log, processGroup)
+		if err != nil {
+			return false, err
+		}
+
+		if needsPVCRemoval && podErr == nil {
+			return true, nil
+		}
+	} else if processGroup.ProcessClass.IsStateful() {
+		log.V(1).Info("Could not find PVC for process group ID",
+			"processGroupID", processGroup.ProcessGroupID)
+	}
+
+	if podErr != nil {
+		log.V(1).Info("Could not find Pod for process group ID",
+			"processGroupID", processGroup.ProcessGroupID)
+		return false, podErr
+	}
+
+	needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log)
+
+	if err != nil {
+		return false, err
+	}
+
+	return needsRemoval, nil
```

Comment:
```suggestion
	return  processGroupNeedsRemovalForPod(cluster, pod, processGroup, log)
```

We can simplify the return logic here.